### DISCUSSION TITLE
extending recursive types

### DISCUSSION THREAD

**Discussion by firleaf** [colinhacks/zod#2265] [Q&A] [answered]
Hi, thanks for the great work!

I'm having an issue with the example on recursive types found in the documentation. Defining recursive types works well, but as soon as I try to extend from a recursive type, TypeScript complains. Here is what TypeScript is saying:
```
Property 'extend' does not exist on type 'ZodType<Category, ZodTypeDef, Category>'
```
Here is my example:
```TypeScript
const baseCategorySchema = z.object({
	name: z.string(),
});

type Category = z.infer<typeof baseCategorySchema> & {
	subcategories: Category[];
};

const categorySchema: z.ZodType<Category> = baseCategorySchema.extend({
	subcategories: z.lazy(() => categorySchema.array()),
});

// This is where the issue occurs
const filmCategorySchema = categorySchema.extend({
	director: z.string(),
});
```

I do understand, that `ZodType` could by something that isn't extendable, but I'm not sure how to tell zod that it is.

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const baseCategorySchema = z.object( {
    name: z.string(),
} )

type FilmCategory = z.infer<typeof baseCategorySchema> & {
    subcategories: FilmCategory[]
    director: string,
}

const filmCategorySchema: z.ZodType<FilmCategory> = baseCategorySchema.extend( {
    subcategories: z.lazy( () => filmCategorySchema.array() ),
    director: z.string(),
} )
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

**Reply by firleaf** [reply]
Ah yes, I guess that works :) Thanks for the quick reply.
